fog/sql_recovery_db/src/lib.rs

```diff
@@ -194,7 +195,13 @@ impl RecoveryDb for SqlRecoveryDb {
         &self,
         key: &CompressedRistrettoPublic,
         start_block: u64,
-    ) -> Result<bool, Error> {
+    ) -> Result<u64, Error> {
+        let highest_known_block_count: u64 = match self.get_highest_known_block_index() {
+            Ok(block_count) => block_count.unwrap_or(0),
+            Err(_) => 0,
+        };
+
+        let accepted_start_block = min(start_block, highest_known_block_count);
```

Comment:
shouldn't this be a max?

Reply:
As discussed offline, yes this should be max.